src/layers/CustomDisplayObjects/ComplexRopeGeometry.ts

```diff
@@ -0,0 +1,212 @@
+import { MeshGeometry } from 'pixi.js';
+import { sum, max } from 'd3-array';
+import { ComplexRopeSegment } from './ComplexRope';
+
+/**
+ * RopeGeometry allows you to draw a geometry across several points and then manipulate these points.
+ * @example
+ * import { RopeGeometry, Point } from 'pixi.js';
+ *
+ * for (let i = 0; i < 20; i++) {
+ *     points.push(new Point(i * 50, 0));
+ * };
+ * const rope = new RopeGeometry(100, points);
+ * @memberof PIXI
+ */
+export class ComplexRopeGeometry extends MeshGeometry {
+  /** An array of segments with points and diameter that determine the rope. */
+  public segments: ComplexRopeSegment[];
+
+  /** Rope texture scale. */
+  public readonly textureScale: number;
```

Comment:
Do these need to be public?

Reply:
If we ever want to reuse Pixi objects when data has changed we need to be able to set segments on an existing ComplexRope. But right now we always throw away all objects and recreate with new calculated data. So making these private for now as suggested.